### DISCUSSION TITLE
Dynamic Column Sorting

### DISCUSSION THREAD

**Discussion by craigmcampbell** [prisma/prisma#7446] [Q&A] [answered]
Hi all, I'm not finding a good way to dynamically sort by a column based on a variable. Say I'm displaying an HTML table and you can sort by one of three headers: id, name, date created.

This is what my query looks like to sort by name:

```ts
return await prisma.templates.findMany({
    where: {
      application_id: applicationId,
    },
    orderBy: [
      {
        name: 'asc',
      },
    ],
  });
```

Is there a way that I'm not seeing to pass in a column to be sorted on dynamically? End result would look something like this:

```ts
const getTemplatesByApplication = async (applicationId: number, sortBy: string) => {
  return await prisma.templates.findMany({
    where: {
      application_id: applicationId,
    },
    orderBy: [
      {
        sortBy: 'asc',
      },
    ],
  });
};
```

Having that same dynamic approach for sort direction would be great too. Thanks!

**Comment by ryands17** [comment, marked as answer]
@craigmcampbell 👋 

You can do the following:

```ts
const getTemplatesByApplication = async (applicationId: number, sortBy: string) => {
  return await prisma.templates.findMany({
    where: {
      application_id: applicationId,
    },
    orderBy: [
      {
        [sortBy]: sortOrder,
      },
    ],
  });
};
```

Where `sortBy` and `sortOrder` are the dynamic values.

**Reply by craigmcampbell** [reply]
I knew I was missing something simple. Thanks so much @ryands17!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
